File: study/lua/src/app/utils/data/TeamDataHelper.lua
--
-- Author: Liangxu
-- Date: 2017-05-27 17:33:39
-- 阵容数据帮助类
local TeamDataHelper = {}

--在某个阵位上是否有某个装备
--pos:阵位
--baseId：装备静态Id
function TeamDataHelper.isHaveEquipInPos(baseId, pos)
	local equipIds = G_UserData:getBattleResource():getEquipIdsWithPos(pos)
	for i, id in ipairs(equipIds) do
		local equipData = G_UserData:getEquipment():getEquipmentDataWithId(id)
		local equipBaseId = equipData:getBase_id()
		if equipBaseId == baseId then
			return true
		end
	end
	return false
end

function TeamDataHelper.getOpenLevelWithId(funcId)
	assert(funcId, string.format("FunctionConst can not find funcId be nil "))

	local config = require("app.config.function_level").get(funcId)
	assert(config, string.format("function_level config can not find id = %d", funcId))
	
	return config.level
end

--获取pos阵位上的武将baseId
function TeamDataHelper.getHeroBaseIdWithPos(pos)
	local heroId = G_UserData:getTeam():getHeroIdWithPos(pos)
	local unitData = G_UserData:getHero():getUnitDataWithId(heroId)
	local baseId = unitData:getBase_id()
	return baseId
end

--获取开启阵位的数量
function TeamDataHelper.getTeamOpenCount()
	local count = 0
	for i = 1, 6 do
		local isOpen = require("app.utils.LogicCheckHelper").funcIsOpened(FunctionConst["FUNC_TEAM_SLOT"..i])
		if isOpen then
			count = count + 1
		end
	end
	return count
end

return TeamDataHelper

File: study/ts/script/utils/data/TeamDataHelper.ts
import { G_UserData, G_ConfigLoader } from "../../init";
import { ConfigNameConst } from "../../const/ConfigNameConst";
import { FunctionConst } from "../../const/FunctionConst";
import { LogicCheckHelper } from "../LogicCheckHelper";
export namespace TeamDataHelper {
    export function isHaveEquipInPos(baseId, pos) {
        let equipIds = G_UserData.getBattleResource().getEquipIdsWithPos(pos);
        for (let i in equipIds) {
            let id = equipIds[i];
            let equipData = G_UserData.getEquipment().getEquipmentDataWithId(id);
            let equipBaseId = equipData.getBase_id();
            if (equipBaseId == baseId) {
                return true;
            }
        }
        return false;
    };
    export function getOpenLevelWithId(funcId) {
        console.assert(funcId, 'FunctionConst can not find funcId be nil ');
        let config = G_ConfigLoader.getConfig(ConfigNameConst.FUNCTION_LEVEL).get(funcId);
        console.assert(config, 'function_level config can not find id = %d');
        return config.level;
    };
    export function getHeroBaseIdWithPos(pos) {
        let heroId = G_UserData.getTeam().getHeroIdWithPos(pos);
        let unitData = G_UserData.getHero().getUnitDataWithId(heroId);
        let baseId = unitData.getBase_id();
        return baseId;
    };
    export function getTeamOpenCount() {
        let count = 0;
        for (let i = 1; i <= 6; i++) {
            let isOpen = LogicCheckHelper.funcIsOpened(FunctionConst['FUNC_TEAM_SLOT' + i])[0];
            if (isOpen) {
                count = count + 1;
            }
        }
        return count;
    };
};